application/extensions/SettingsWidget/SettingsWidget.php

```diff
@@ -614,7 +614,7 @@ public function renderDate($name, array $metaData, $form = null)
         $value = isset($metaData['current']) ? $metaData['current'] : '';
         $html = Yii::app()->getController()->widget('yiiwheels.widgets.datetimepicker.WhDateTimePicker', array(
                 'name' => $name,
-                'id' => $name,
+                'id' => \CHtml::getIdByName($name),
```

Comment:
I was looking at this. This is the only widget it has the id specifically set. I think that if the id attribute is removed, Yii will use getIdbYName anyway...

Reply:
I think it's only since a YII update, removing the id here : we get `yw32`


1. Look at HTML for text, texarea, dropddown …: they have their own id
2. Check label : use `getIdByName` (and it's needed)
3. With this good id : label action work (see peek)
4. 

![Peek -NOID](https://user-images.githubusercontent.com/1439428/192445796-7e3c298c-40d2-4e4d-be97-ca77f916e374.gif)
![Peek -GOODID](https://user-images.githubusercontent.com/1439428/192445800-4456363f-2563-4992-8b45-e59fcbca8593.gif)

Reply:
PS : https://github.com/LimeSurvey/LimeSurvey/commit/45ed548e32c742f6a01bca9641353b09e7925409

Reply:
You're right : https://github.com/LimeSurvey/LimeSurvey/blob/e058b839f77043eb10a8d366d4bd69ae74014633/third_party/yiisoft/yii/framework/web/helpers/CHtml.php#L890 

See : https://github.com/yiisoft/yii/commit/0154c5e64872a5e4112f45dacb2f5ff645f1db61

Reply:
OK, let me know if you want to keep it like this or change it. Thanks!

Reply:
What do you mean ? 

This fix the issue, t andhis is needed because `'yiiwheels.widgets.datetimepicker.WhDateTimePicker` use a bad id.

Reply:
You're right about CHtml fro Yii core, not `yiiwheels` widget :)